### PR TITLE
feat(storage): Firebase JS SDK v9 modular API

### PR DESCRIPTION
**Pull request by russellwheatley**
### Description

See title.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [ ] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ ] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Review by Lyokone — DISMISSED** [review]

File: docs/database/usage/index.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ import database from '@react-native-firebase/database';
 const reference = database().ref('/users/123');
 ```
 
-NOTE: To get a reference to a database other than an 'us-central1' default database, you must pass the database URL. You can find your Realtime Database URL in the Realtime Database section of the Firebase console.
+NOTE: To get a reference to a database other than a 'us-central1' default database, you must pass the database URL. You can find your Realtime Database URL in the Realtime Database section of the Firebase console.
```
Comment (mikehardy) [inline docs/database/usage/index.md:53]: Strange, I fixed this on main already, why is this showing up in a diff here? 🤔

File: tests/app.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ import '@react-native-firebase/database';
 import '@react-native-firebase/dynamic-links';
 import '@react-native-firebase/firestore';
 import * as functionsModular from '@react-native-firebase/functions';
+import * as storageModular from '@react-native-firebase/storage';
```
Comment (mikehardy) [inline tests/app.js:32]: Really minor point - but can you snuggle this one up under the non-modular storage import just a few lines below? These imports have been re-positioned a couple times and right now we have it organized such that the same module is imported modular/non-modular as a unit?

**Review by mikehardy — CHANGES_REQUESTED** [review]
This looks really good - I had two really minor thoughts for the changes implemented, trivial really but need a look

Main structural thought is that we've had one of these modular PRs go in with a delete of the non-modular tests, since the modular e2e *includes* the compatibility testing, that actually seems like a viable path forward, and reduces e2e time. We have also had a few go in like this where the e2e non-modular tests are effectively duplicated, with no delete of the original tests.

What do you prefer here? I lean towards a delete of the non-modular tests personally but it is a weakly held opinion.

**Review by mikehardy — APPROVED** [review]
LGTM! Thanks - another one bites the dust, and this was medium-large IMHO

**Pull request merged**